pkg/deploy/controller/image_change_controller.go

```diff
@@ -48,7 +49,8 @@ func (c *ImageChangeController) HandleImageRepo() {
 		for _, trigger := range config.Triggers {
 			if trigger.Type == deployapi.DeploymentTriggerOnImageChange &&
```

Comment:
split this into two ifs - the first if should be "not trigger on image change || not automatic" continue.  Make this easier to read.